generate/generate.go

```diff
@@ -158,44 +141,75 @@ func Do(root string) error {
 	return nil
 }
 
-func generateStackLocals(
-	rootdir string,
+func writeStackLocalsCode(
+	root string,
 	stackpath string,
-	metadata terramate.StackMetadata,
+	stackMetadata terramate.StackMetadata,
 	globals *terramate.Globals,
+	targetLocalsFile string,
 ) error {
 	logger := log.With().
-		Str("action", "generateStackLocals()").
+		Str("action", "writeStackLocalsCode()").
+		Str("root", root).
 		Str("stack", stackpath).
+		Str("targetLocalsFile", targetLocalsFile).
 		Logger()
+	logger.Debug().Msg("Save stack locals.")
 
-	logger.Trace().
-		Msg("Get generated file path.")
-	genfile := filepath.Join(stackpath, LocalsFilename)
-	if err := checkFileCanBeOverwritten(genfile); err != nil {
-		return err
+	stackLocalsCode, err := generateStackLocalsCode(root, stackpath, stackMetadata, globals)
+	if err != nil {
+		err = errutil.Chain(ErrExportingLocalsGen, err)
+		return fmt.Errorf("stack %q: %w", stackpath, err)
```

Comment:
I see you're spliting the err return multiple times, maybe if you create a helper alias it improves to keep it one line:
```suggestion
		return errorf("stack %q: %w", stackpath, errchain(ErrExportingLocalsGen, err))
```
but feel free to ignore also, just a thought.

Reply:
this stayed on multiple lines more because of the previous code dealing with err lists. wil fix.